src/hooks/useMultiColumnListSort.test.js

```diff
@@ -0,0 +1,62 @@
+import {
+  renderHook,
+  act,
+} from '@testing-library/react-hooks';
+
+import useMultiColumnListSort from './useMultiColumnListSort';
+
+import { sortOrders } from '../constants';
+
+describe('Given useMultiColumnListSort', () => {
+  it('should return default values', () => {
+    const currentParameters = {
+      sortOrder: '',
+      sortedColumn: '',
+    };
+    const { result } = renderHook(() => useMultiColumnListSort(currentParameters.sortOrder, currentParameters.sortedColumn));
+    const [sortParameters] = result.current;
+
+    expect(sortParameters.sortOrder).toBe('');
+    expect(sortParameters.sortedColumn).toBe('');
+  });
+
+  describe('when handle "onChange" action', () => {
+    it('should return ascending sortOrder and selected sortedColumn', () => {
+      const currentParameters = {
+        sortOrder: {
+          name: 'asc',
+          fullName: 'ascending',
+        },
+        sortedColumn: 'newName',
+      };
+      const { result } = renderHook(() => useMultiColumnListSort(currentParameters.sortOrder, currentParameters.sortedColumn));
+      const [sortParameters, onChange] = result.current;
+
+      act(() => {
+        onChange('', { name: 'newName' });
+      });
+
+      expect(sortParameters.sortOrder).toEqual(sortOrders.asc);
```

Comment:
Looking at code this hook should change sort from `asc` to `desc` and vice versa if `onChanged` is called with same name parameter, but this test expects to get the same `sortOrder` as before `onChange` and it passes.
The issues is that after `onChange` we need to get updated data in `result`. This can be done by awaiting `waitForNextUpdate`
```
const { result, waitForNextUpdate } = renderHook(...);
...
act(() => {
  onChange(...);
});

await waitForNextUpdate();

expect(result.current[0].sortOrder).toEqual(sortOrders.desc);
```